### PR TITLE
add delay-prediction-service

### PR DESCRIPTION
**Pull request by derhuerst**

### CODE REVIEW

File: roles/delay-prediction-service/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+- name: Install systemd files
+  template:
+    src: '{{ item }}'
+    dest: /etc/systemd/system/
+  with_items:
+    - delay-prediction-service.service
+  notify: Restart delay-prediction-service
+
+- todo: create PostgreSQL database
```
Comment (derhuerst) [inline roles/delay-prediction-service/tasks/main.yml:10]: An empty database is need here that the build script will put data in.

File: roles/delay-prediction-service/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- todo: run build script
```
Comment (derhuerst) [inline roles/delay-prediction-service/tasks/main.yml:11]: The build script (`./build.sh`) also needs to be run.

File: roles/delay-prediction-service/templates/delay-prediction-service.service [hunk 3]
```diff
@@ -0,0 +1,34 @@
+[Unit]
+Description=pull bus positions from Thingsboard, predict delays & positions, send GTFS-RT to MQTT
+After=network-online.target
+Wants=network-online.target
+
+[Service]
+Restart=always
+RestartSec=5
+
+Environment=THINGSBOARD_HOST="{{ thingsboard_api_url }}"
```
Comment (derhuerst) [inline roles/delay-prediction-service/templates/delay-prediction-service.service:10]: I can decrypt the secrets. Can you check if `thingsboard_api_url` is just the host (e.g. `thingsboard.cloud`) or a full URL (e.g. `https://thingsboard.cloud/api/foo`)?

File: roles/delay-prediction-service/templates/delay-prediction-service.service [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Environment=THINGSBOARD_USERNAME="{{ thingsboard_username }}"
+Environment=THINGSBOARD_PASSWORD="{{ thingsboard_password }}"
+Environment=THINGSBOARD_DEVICE_GROUP="f1740240-7b3f-11eb-970d-837f8b4d738d"
+Environment=MQTT_URI="{{ mqtt_uri }}"
+Environment=PGUSER="{{ todo }}"
```
Comment (derhuerst) [inline roles/delay-prediction-service/templates/delay-prediction-service.service:15]: Environment variables necessary for accessing PostgreSQL are necessary here.

File: roles/delay-prediction-service/tasks/main.yml (hunk 1 above)
Comment (leonardehrenfried) [reply roles/delay-prediction-service/tasks/main.yml:10]: Just Posgres or PostGis?

File: roles/delay-prediction-service/templates/delay-prediction-service.service (hunk 3 above)
Comment (leonardehrenfried) [reply roles/delay-prediction-service/templates/delay-prediction-service.service:10]: It's a full URL:

```
Environment=THINGSBOARD_HOST="https://portal.mhascaro.com/api"
```

File: roles/delay-prediction-service/tasks/main.yml (hunk 1 above)
Comment (derhuerst) [reply roles/delay-prediction-service/tasks/main.yml:10]: PostGIS

**Pull request merged**